server/auth.go

```diff
@@ -557,6 +578,54 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 		err  error
 		ao   bool // auth override
 	)
+
+	// Check if we have auth callouts enabled at the server level or in the bound account.
+	defer func() {
```

Comment:
This is a lot of security-critical work to be happening inside a deferred function.  I'm worried about this being triggered after a panic.  Is this something which could be put into a new intermediary function instead?

Reply:
This needs to be a defer to be able to reference the result of authorized which is a return variable.